linux/setup/test_setup.yml

```diff
@@ -0,0 +1,39 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Below tasks will be executed at the beginning of each Linux test case.
+# If base snapshot does not exist, will take a snapshot of VM as the
+# base snapshot, if it exists, then revert to it directly.
+#
+- name: "testcase: {{ ansible_play_name }}"
+  debug:
+    msg: "Starting testcase {{ ansible_play_name }}"
+
+- name: Set fact of the info collection folder of test case
+  set_fact:
+    testcase_info_collect_path: "{{ testrun_log_path }}/{{ ansible_play_name }}"
+
+- include_tasks: base_snapshot_check_revert.yml
+
+- include_tasks: vm_guest_ip_get_save.yml
+
+# Only get vmtools status when necessary
+- include_tasks: ../../common/vm_get_vmtools_status.yml
+  when:
+    - vmtools_check | default(True)
+    - use_saved_base_ip | bool
+
+# Get guest system info once
+- include_tasks: ../../common/get_guest_system_info.yml
+
+- include_tasks: ../utils/check_guest_os_gui.yml
+  when: guest_os_with_gui is undefined
```

Comment:
How about put setting guest_os_with_gui in common/get_guest_system_info.yml since it is also system info? The other changes look good to me.

Reply:
Add 2 new tasks `linux/utils/get_linux_system_info.yml` and `windows/utils/get_windows_system_info.yml`, so we can add Linux or Windows specific info into these separate files.